src/iap/shared/components/carousel/carousel.component.ts

```diff
@@ -0,0 +1,63 @@
+import { Component, Input, OnDestroy, OnInit, ViewChild } from '@angular/core';
+import { Slide, SlideComponent } from '@shared/models/carousel.models';
+import { SlideDirective } from '../../directives/slide.directive';
+
+@Component({
+	selector: 'iap-carousel',
+	templateUrl: './carousel.component.html',
+	styleUrls: ['./carousel.component.scss'],
+})
+export class CarouselComponent implements OnInit, OnDestroy {
+	@ViewChild(SlideDirective, { static: true }) slideHost!: SlideDirective;
+
+	@Input()
+	slides: Slide<any>[] = [];
+
+	currentSlideIndex = 0;
+
+	interval: NodeJS.Timeout | undefined;
+
+	ngOnInit(): void {
+		this.selectSlide(this.currentSlideIndex);
+		this.startSlideshow();
+	}
+
+	ngOnDestroy() {
+		clearInterval(this.interval);
+	}
+
+	onNextButtonClick() {
+		this.restartSlideshowTimer();
+
+		this.selectSlide(this.currentSlideIndex + 1);
+	}
+
+	onPrevButtonClick() {
+		this.restartSlideshowTimer();
+
+		if (this.currentSlideIndex > 0) this.selectSlide(this.currentSlideIndex - 1);
+		else this.selectSlide(this.slides.length - 1);
```

Comment:
I'm not sure about backend code style, but frontend developers (as I noticed) try to avoid inline selection statements using.
```suggestion
		if (this.currentSlideIndex > 0) {
		  this.selectSlide(this.currentSlideIndex - 1);
		} else {
		  this.selectSlide(this.slides.length - 1);
		}
```
But. If you really want to short this expression, you can use the next code part:
```suggestion
		this.selectSlide((this.currentSlideIndex || this.slides.length) - 1);
```
Or:
```suggestion
		this.selectSlide(this.currentSlideIndex > 0 ? this.currentSlideIndex - 1 : this.slides.length - 1);
```